flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/SAI/A320_Neo_SAI.js

```diff
@@ -1125,13 +1125,30 @@ class A320_Neo_SAI_Brightness extends NavSystemElement {
         this.maximum = 0.99;
         this.minimum = 0.15;
         this.bright_gran = 0.05;
+        this.dayBrightness = 0.85;
+        this.nightBrightness = 0.50;
+        this.isDaytime = this.getIsDaytime();
+        this.currentBrightness = this.isDaytime ? this.dayBrightness : this.nightBrightness;
+        this.isInTransition = false;
+        this.transitionSpeed = 5e-5;
+        this.updateThrottler = new UpdateThrottler(10000);
     }
     onEnter() {
     }
     isReady() {
         return true;
     }
     onUpdate(_deltaTime) {
+        if (!this.isInTransition && this.updateThrottler.canUpdate(_deltaTime) === -1) {
+            return;
+        }
+
+        if (this.isDaytime !== this.getIsDaytime()) {
+            this.isDaytime = this.getIsDaytime();
+            this.isInTransition = true;
+        }
+
+        this.updateTransition(_deltaTime);
```

Comment:
You should never ever overrule the updateThrottler.

this.updateTransitions seems to only have one use case so I don't see the need to create a new method for it.

Generally there is one key issue, the "day" brightness can be reduced below the "night" brightness, so the brightness may increase on a day->night transition.

I am still looking further into it..

Reply:
I've taken a look at how to solve these issues, I've made a pr to your fork with a suggestion on how to solve the issue.
https://github.com/BlueberryKing/a32nx/pull/1

Reply:
Thank you for your effort! I will keep the correct usage for `updateThrottler` in mind. Your changes look good. The only thing I've noticed as a side effect of not having the "reverse" brightness transitions is that the brightness does not go back to your set presets, so if you increase the brightness at night from 0.5 to 0.7, then it turns daytime, so brightness is 0.85, but after it turns night again, it will go back to 0.5, not 0.7

To be honest, this is quite a rare scenario and I'm not even sure it's necessarily a bad thing. 

I think we should have some people test it with the goal of the system working without them ever having to touch it, possibly using the buttons in case of clouds or something.

Reply:
Well if this actually is a thing, we still can use the diff to calculate the offset and adopt it to the new target brightness anyways...

So basically apply the brightness changes to the new "preset" like this?
Well, I kind of like that idea, ngl.

```js
this.targetBrightness = isDaytime ?
    Math.max(Math.min(this.maximum, this.targetBrightness * 2 - this.nightBrightness), this.dayBrightness) : 
    Math.min(Math.min(this.maximum, this.targetBrightness * 2 - this.dayBrightness), this.nightBrightness);
```

Reply:
Some of those `Math.max`, `Math.min` and `this.maximum` look wrong. Now, I'm not sure what you're proposing 😄. To implement what I'm saying, we would probably need to modify `this.dayBrightness` and `this.nightBrightness` at some point just to save the "preset" value somewhere. Right now, these values are (purposely) never mutated.

We just need to make sure that, if, for example, `this.dayBrightness` were to drop below `this.nightBrightness` through manual adjustment, we also "push" `this.nightBrightness` down.

We might be overengineering this a bit, but I'll think about it some more.

Reply:
That's exactly what my proposal does 😅 it pushes the offset to the next "phase" and maintains the difference while also ensuring we are not exceeding the maximum brightness,that's what the "inner" Math.Max is for.

Reply:
Alright this is the switcher, it will ensure the corrected value is within min max range and will adjust for the difference

The switch happens by multiplying the difference of day and night normal values (.45) with 1 or -1 depending on the transition.

```js
this.targetBrightness = Math.max(this.minimum, Math.min(this.maximum, this.targetBrightness - (this.nightBrightness - this.dayBrightness) * (isDaytime ? 1 : -1)));
```

Reply:
Ah, now this looks like what I had in mind with the offset to the presets. I think there is still a small inconsistency in this scenario:
At night, you manually adjust from 0.5 to 0.8, so offset is 0.3. During the automatic transition to daytime, the value gets clamped from (0.85 + 0.3 = 1.15) to 0.99. Now the offset is only 0.99 - 0.85 = 0.14, so on the next nighttime transition, brightness will be 0.5 + 0.14 = 0.69, not 0.8 as you originally set it.

Admittedly, this is a very constructed example and not necessarily unexpected behavior.

Reply:
Keep in mind that 0.3 means that you've increased the brightens by 30% (and that's in the night) and end up with a brightness 5% less than daylight brightness. (with this setting the isis basically is a flashbang)
I don't really see this scenario being realistic.

I'd say lets add the dynamic change and let people play around with it during QA and gather feedback.
We can still adjust the behavior afterwards with a new pr if people dislike the behavior.

Reply:
Fair enough, sounds good 👍.